app/assets/javascripts/editorToolbar.js

```diff
@@ -108,21 +108,17 @@ $(function() {
           switch (file_type) {
             case 'image':
               orig_image_url = file_url + '?s=o' // size = original
-              $E.wrap('[![', '](' + file_url + ')](' + orig_image_url + ')', {'newline': true, 'fallback': data.result['filename']}) // on its own line; see /app/assets/js/editor.js
+              $E.wrap('[![', '](' + file_url + ')](' + orig_image_url + ')', true, data.result['filename']);
               break;
             case 'csv':
-              $E.wrap('[graph:' + file_url + ']', '', {'newline': true})
+              $E.wrap('[graph:' + file_url + ']', '', true);
               break;
             default:
-              $E.wrap('<a href="'+data.result.url.split('?')[0]+'"><i class="fa fa-file"></i> ','</a>', {'newline': true, 'fallback': data.result['filename'].replace(/[()]/g , "-")}) // on its own line; see /app/assets/js/editor.js
```

Comment:
Why was fallback used? and why is it removed now?

Reply:
This is the only place where `fallback` is still used (3 times):
https://github.com/publiclab/plots2/blob/97247e30289dfb27f93446b84d036ad9811e7983/app/assets/javascripts/editorToolbar.js#L108-L118

I didn't delete `fallback` here, but I just separated it out into two parameters instead of one object:
- **Before**: `$E.wrap(a, b, { fallback: "fallback", newline: "true" });`
- **After**: `$E.wrap(a, b, true, "fallback")`;`

In other words, `wrap` now takes four parameters, instead of three. And none of the parameters are objects.

I know it might be a little harder to read but I really needed to do this to pass codeclimate. It was failing this PR saying that the old code in `$E.wrap` had too much 'cognitive complexity'. Part of the reason why was because the parameter was an object, and we had to do a check like `const fallBackExists = args && args['fallback']`&mdash;which adds to cognitive complexity.

**tldr;** I didn't delete fallback, the same functionality is still there. Just changed parameters.